src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6585,8 +6586,13 @@ public override IOperation VisitInstanceReference(IInstanceReferenceOperation op
                     }
                     else
                     {
-                        Debug.Assert(operation.Parent is InvocationOperation { Parent: CollectionExpressionOperation ce } && ce.HasErrors(_compilation),
-                            "Expected to reach this only in collection expression infinite chain cases.");
+                        Debug.Assert(
+                            // We have a collection expression infinite chain case
+                            (operation.Parent is InvocationOperation { Parent: CollectionExpressionOperation ce } && ce.HasErrors(_compilation)) ||
+                            // We have an object initializer with an invalid assignment
+                            (operation.Parent is { Kind: OperationKind.None }),
```

Comment:
>(operation.Parent is { Kind: OperationKind.None }) [](http://example.com/codeflow?start=28&length=50)

Similar comment about lack of illustration in tests.

Reply:
Reverted